### ISSUE TITLE
Checks/Downloads hang if the title/filename has an emoji (unicode)

### ISSUE THREAD

**Issue by kdliu86** [axcore/tartube#106]
This only happens with TarTube; I tried downloading using only youtube-dl and it worked fine.

Ex. 

"10 Best Love Songs To Play On Guitar 🎸 ❤️ - GuitarZero2Hero"
https://www.youtube.com/watch?v=RsMP0ZssMs4

`youtube-dl --newline -i --hls-prefer-native -o E:/TarTube-Downloads/Unsorted Videos/%(title)s-%(id)s.%(ext)s --dump-json --verbose https://www.youtube.com/watch?v=RsMP0ZssMs4
[debug] Default format spec: bestvideo+bestaudio/best
[Unsorted Videos] <Simulated download of: '10 Best Love Songs To Play On Guitar 🎸 ❤️ - GuitarZero2Hero-RsMP0ZssMs4'>`

(Nothing happens after this is outputted).

**Comment by kdliu86** [comment]
NM I forced a manual update of Youtube-DL and it worked to fix the hanging issue..

**Comment by kdliu86** [comment]
Ok I just tried this again,

I cannot download Playlists/Channels with emojis in them, though individual downloads work fine

ex:

https://www.youtube.com/watch?v=RsMP0ZssMs4

vs 

https://www.youtube.com/channel/UCasFZzbM8JJ6dqSVEgL9VVg/videos

**Comment by kdliu86** [comment]
Additional example here, on a check channel, the thread hangs when it loads the heart emoji title.

```
youtube-dl --newline -i --hls-prefer-native -o E:/TarTube-Downloads/YT/BadBunny/%(title)s-%(id)s.%(ext)s --dump-json --download-archive E:/TarTube-Downloads/YT/BadBunny/ytdl-archive.txt https://www.youtube.com/channel/UCmBA_wu8xGg1OfOkfW13Q0Q/videos
[BadBunny] <Simulated download of: 'SI ELLA SALE - Bad Bunny _ Las Que No Iban A Salir-L88MenfjtTI'>
[BadBunny] <Simulated download of: 'MÁS DE UNA CITA - Bad Bunny x Zion y Lennox _ Las Que No Iban A Salir-UkC78ZKHn_k'>
[BadBunny] <Simulated download of: 'BYE ME FUI - Bad Bunny _ Las Que No Iban A Salir-Ts_pOxNJtVg'>
[BadBunny] <Simulated download of: 'CANCIÓN CON YANDEL - Yandel x Bad Bunny _ Las Que No Iban A Salir-R82Pk0VQWOQ'>
[BadBunny] <Simulated download of: 'PA' ROMPERLA - Bad Bunny x Don Omar _  Las Que No Iban A Salir-tK4ioPhpal4'>
[BadBunny] <Simulated download of: 'BAD CON NICKY - Bad Bunny x Nicky Jam _ Las Que No Iban A Salir-0oZYC1k6ufs'>
[BadBunny] <Simulated download of: 'BENDICIONES - Bad Bunny _ Las Que No Iban A Salir-WNadJHcL0X0'>
[BadBunny] <Simulated download of: 'CÓMO SE SIENTE (Remix) - Jhay Cortez x Bad Bunny _ Las Que No Iban A Salir-ZAW9358S9rY'>
[BadBunny] <Simulated download of: 'RONCA FREESTYLE - Bad Bunny _ Las Que No Iban A Salir-c6vb2uJ5QBU'>
[BadBunny] <Simulated download of: 'EN CASITA feat. Gabriela - Bad Bunny _ Las Que No Iban A Salir-JW1a_m14UkI'>
[BadBunny] <Simulated download of: 'Yo Perreo Sola - Bad Bunny ( Video Oficial )-GtSRKwDCaZM'>
[BadBunny] <Simulated download of: 'Bichiyal - Bad Bunny x Yaviah ( Video Oficial )-Udxl17LVHYA'>
[BadBunny] <Simulated download of: 'Hablamos Mañana ( VHS ) - Bad Bunny x Duki x Pablo Chill-E-Da-CIlJU8I4'>
[BadBunny] <Simulated download of: 'Hablamos Mañana - Bad Bunny x Duki x Pablo Chill-E ( Video oficial )-j2FNs-hX93E'>
[BadBunny] <Simulated download of: 'Pero Ya No - Bad Bunny ( Video Oficial )-bg82FqrCjcs'>
[BadBunny] <Simulated download of: 'Si Veo a Tu Mamá - Bad Bunny ( Video Oficial )-CPK_IdHe1Yg'>
[BadBunny] <Simulated download of: 'La Difícil - Bad Bunny (Video Oficial ) _ YHLQMDLG-fEYUoBgYKzw'>
[BadBunny] <Simulated download of: 'Soliá - Bad Bunny _ YHLQMDLG-bW4nKN0QL5A'>
[BadBunny] <Simulated download of: '🖤 - Bad Bunny _ YHLQMDLG-fSlqTX39CMM'>
```

**Comment by axcore** [comment]
I'm not sure what might be causing your issue. I have never experienced any problems downloading videos or channels whose names contain unicode characters. Indeed, I was able to do **Check All**  on the [link just above](https://www.youtube.com/channel/UCmBA_wu8xGg1OfOkfW13Q0Q/videos), and I didn't experience any delays:

![delay1](https://user-images.githubusercontent.com/37951926/83530877-3e1fed80-a4e4-11ea-922d-3217b5a2eb02.png)
![delay2](https://user-images.githubusercontent.com/37951926/83530885-3f511a80-a4e4-11ea-8dc8-26a4394a39d1.png)

**Comment by kdliu86** [comment]
I am on Windows 10 Pro v1909, Python 3.8, YouTube-dl v2020.5.29

**Comment by axcore** [comment]
Yes, that's possible. Tartube doesn't use quotes because Python sends each argument/option to the OS as a discrete unit (the OS already knows that "My favourite video" is a single argument, not three).

I will test it on MS Windows, with quotes, and see what happens.

**Comment by profucius** [comment]
I am experiencing this issue as well. A number of channels I follow choose to use unicode emoji in their video titles. This causes Tartube to stall when parsing that video, and it will not proceed to check further videos in that channel. I am forced to set Download Options to scan videos prior to the date of the video the emoji title video exists. It will proceed, until it hits another emoji. Again, some channels use emoji in many videos, making Tartube completely unable to manage those channels.

I am on Windows 10 Pro v1909, Python 3.8, YouTube-dl v2020.6.16.1

As OP stated, I also have no issue using youtube-dl command line to download videos with emoji unicode in the title. It seems to only be Tartube on Windows that has the issue.

Is there a way to have Tartube simply skip a video if it hangs on it while checking? This would at least get us going again for now, until you're able to resolve the underlying issue with unicode detection on the Windows version.

**Comment by axcore** [comment]
I have tried modifying the youtube-dl system command:

1. Adding quotation marks around file paths breaks everything; no videos/channels/playlists can be checked or downloaded
2. Sending the system command as a single string, rather than a list of arguments, breaks everything

>Is there a way to have Tartube simply skip a video if it hangs on it while checking?
No; youtube-dl operates independently of Tartube. We can start and stop it, but we can't change anything once it is running.

I thought about adding a timeout for each checked video, but in fact a 60 second timeout already exists and is enabled by default (Edit > System preferences... > youtube-dl > When checking videos, apply a 60-second timeout).

I don't know what else to suggest. I cannot reproduce your problem and I cannot think of any way to solve it.

**Comment by profucius** [comment]
I tried to see if Tartube would skip on a 60 second timeout, but it does not. I do have the setting enabled. I recorded a screen cap of this being demonstrated.

![image](https://user-images.githubusercontent.com/18254607/86515264-b3355a00-bde5-11ea-9c0c-877c3a705de2.png)


![tartube-unicode](https://user-images.githubusercontent.com/18254607/86515270-b597b400-bde5-11ea-849f-136e3b7b4224.gif)

**Comment by axcore** [comment]
I'm stumped, then. It's so difficult to fix these problems that I can't even reproduce...

**Comment by profucius** [comment]
Is there an application log setting that I can turn on that might explain what the app is doing when it stalls out?

Interestingly, I found if I use the classic mode downloader that it will download the files, no stall. Maybe it is just the simulated download? Could it be when Tartube is trying to grab the thumbnail or the description file? Since classic mode only grabs the video.

**Comment by axcore** [comment]
Possibly. You can turn on debug messages by editing the source code, found in `C:\Users\YOURNAME\AppData\Local\Tartube\msys64\home\user\tartube\tartube`

In the mainapp.py file, line 93 (in v2.1.051); in mainwin.py, line 58, and in downloads.py, line 60. Change the following lines:

`DEBUG_FUNC_FLAG = False
`
to:

`DEBUG_FUNC_FLAG = True
`
The True argument must be capitalised.

You then need to run Tartube from a terminal window. Run `C:\Users\YOURNAME\AppData\Local\Tartube\msys64\mingw64.exe`. In that terminal window, type


```
cd tartube
python3 tartube/tartube
```

**Comment by profucius** [comment]
Apologies for the late response, I'd planned to follow your instructions earlier but was busy with other things at the time.

I followed the above, set the flag to True in all 3 files. However when I run mingw64.exe in cmd, and at the bash I type "cd tartube" it tells me

`$ cd home
-bash: cd: home: No such file or directory
`

**Comment by axcore** [comment]
I did not run minwg64.exe in cmd; I navigated into the folder and double-clicked the mingw64.exe icon.

I just tested this on a clean Windows install; it definitely works:

![windows2](https://user-images.githubusercontent.com/37951926/94988709-26767580-0567-11eb-8b7f-b1587c431a7d.png)

By the way, since v2.2.0 it is no longer necessary to edit the source code. To enable logging, click Edit > System preferences... > General > Debugging, and select the checkbuttons.
